File: 1_two_sum/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(len(nums)):
                if i != j and (nums[i] + nums[j]) == target:
                    return [i, j]

    # Hash Tables - Uses more Memory
    def twoSum_optimzed(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for i, num in enumerate(nums, start=0):
            complement = target - num
            if complement in num_map:
                return [num_map[complement], i]
            num_map[num] = i
        return []

teste = Solution()
print(teste.twoSum_optimzed(nums=[3,3], target=6))

File: 1_two_sum/main.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    for idx_1, i := range nums {
		for idx_2, j := range nums {
			if idx_1 != idx_2 && (i + j) == target {
				return []int{idx_1, idx_2}
			}
		}
	}
	return []int{0, 0}
}

func main() {
	nums := []int{3,3}
	result := twoSum(nums, 6)
	fmt.Println(result)
}